### PR TITLE
Setup single dependency mode

### PR DESCRIPTION
**Pull request by cltnschlosser**
Adapted from: https://github.com/apple/swift/commit/2a79331ac3c26027cf158adfc716e717ad579478

Fixes validation-test/Driver/batch_mode_size_limit.swift

### CODE REVIEW

File: Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift [hunk 1]
```diff
@@ -242,11 +242,23 @@ extension Driver {
...
-        addOutputOfType(
+        if let singleDependenciesFileInput = singleDependenciesFileInput {
+          if input == singleDependenciesFileInput.file {
+            addOutputOfType(
+              outputType: .dependencies,
+              finalOutputPath: dependenciesFilePath,
+              input: input,
+              flag: "-emit-dependencies-path")
+          } else {
+            // FIXME: create empty dummy file
```
Comment (cltnschlosser) [inline Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift]: How should I handle the driver creating files?

File: Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift [hunk 2]
```diff
@@ -242,11 +249,27 @@ extension Driver {
...
             flag: "-emit-module-doc-path")
-        addOutputOfType(
+        if let singleDependenciesFileInput = singleDependenciesFileInput {
+          if input == singleDependenciesFileInput.file {
+            addOutputOfType(
+              outputType: .dependencies,
+              finalOutputPath: dependenciesFilePath,
+              input: input,
+              flag: "-emit-dependencies-path")
+            // FIXME: This should probably support relative paths
```
Comment (cltnschlosser) [inline Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift]: Can take advantage of the work in #73 to make a `writeFileContents(VirtualPath)` once that is merged.

**Comment by DougGregor** [comment]
Uh oh. We were thinking of removing this feature entirely from the old Swift driver, and instead having the build systems only ask for a single dependency file.

**Comment by cltnschlosser** [comment]
> Uh oh. We were thinking of removing this feature entirely from the old Swift driver, and instead having the build systems only ask for a single dependency file.

Haha, maybe we just let this one sit then. Or close now if we are sure this feature is going to be removed. I’ve just been going through the lit test failures and picking files that sound interesting.

**Comment by DougGregor** [comment]
I'm fine with letting it sit until we're sure, thanks

**Comment by cltnschlosser** [comment]
Should I pull out the driver show lifecycle stuff into a separate PR?

**Comment by DougGregor** [comment]
> Should I pull out the driver show lifecycle stuff into a separate PR?

Yes, please!

**Comment by shahmishal** [comment]
The Swift project moved the default branch to `main` and deleted `master` branch, so GitHub automatically closed the PR. Please re-create pull request with `main` branch.

More detail about the branch update - https://forums.swift.org/t/updating-branch-names/40412

**Pull request closed without merging**